jgrapht-core/src/main/java/org/jgrapht/alg/lca/EulerTourRMQLCAFinder.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.lca;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.LCAAlgorithm;
+
+import java.util.*;
+
+import static org.jgrapht.util.MathUtil.log2;
+
+/**
+ * Algorithm for computing lowest common ancestors in rooted trees and forests using the Euler tour approached
+ * introduced in <i>Berkman, Omer; Vishkin, Uzi (1993), "Recursive Star-Tree Parallel Data Structure",
```

Comment:
This sentence is not clear. Is the Euler tour approach introduced in this paper or the LCA technique using the Euler tour approach? If the first then correct the reference to the FOCS paper (1984) or the corresponding journal that introduced the approach. Otherwise, make the it clear that you are referring to the LCA technique and not the Euler tour technique.